lib/Runtime/Language/ProfilingHelpers.cpp

```diff
@@ -337,14 +353,34 @@ namespace Js
             }
         } while(false);
 
-        JavascriptOperators::OP_SetElementI(base, varIndex, value, scriptContext, flags);
+        RecyclableObject* cacheOwner;
+        PropertyRecordUsageCache* propertyRecordUsageCache;
+        TypeId instanceType = JavascriptOperators::GetTypeId(base);
+        bool isTypedArray = (instanceType >= TypeIds_Int8Array && instanceType <= TypeIds_Float64Array);
```

Comment:
Why is this condition different than the one in ValueType::IsTypedArray? And why do we care here whether it's a typed array?

Reply:
I'm not sure why the conditions are different, but typed arrays have different semantics than other objects for OP_SetElementI.